models/data.model.js

```diff
@@ -27,3 +27,23 @@ exports.fetchArticles = (id) => {
       return article;
     });
 };
+
+exports.fetchArticles = () => {
+  return db
+    .query(
+      `SELECT articles.article_id, articles.author, articles.title, articles.topic, articles.created_at, articles.votes, articles.article_img_url, CAST(COUNT(comments.article_id)AS INT) AS comment_count
+      FROM comments
+      RIGHT JOIN articles
+      ON articles.article_id = comments.article_id
+      GROUP BY articles.article_id, articles.author, articles.title, articles.topic, articles.created_at, articles.votes, articles.article_img_url
+      ORDER BY created_at DESC;`
+    )
+    .then((result) => {
+      if (result.rows.length > 0) {
+        const articles = result.rows;
+        return articles;
+      } else {
+        return Promise.reject({ status: 404, msg: "Invalid request" });
```

Comment:
404 would be not found

400 is invalid request

Reply:
Im also not 100% what test this is passing or when this would happen. I think you dont actually need this.